data/transactions/logic/eval.go

```diff
@@ -1924,11 +1924,14 @@ func opSetBit(cx *evalContext) {
 		// we're thinking of the bits in the byte itself as
 		// being big endian. So this looks "reversed"
 		mask := byte(0x80) >> bitIdx
+		// Copy to avoid modifying shared slice
+		scratch := append([]byte(nil), target.Bytes...)
```

Comment:
Be aware that 
b = make([]T, len(a))
copy(b, a)
Would be marginally faster. Looks good otherwise.